### PR TITLE
[WIP] Normalize path - remove //

### PR DESCRIPTION
**Pull request by gauravdesale**
wanting to help out with issue #8862 would be great to get assigned to it

### CODE REVIEW

**Comment by justinmk** [comment]
GH doesn't allow assignment to new contributors :( But, consider yourself assigned.

File: src/nvim/path.c [hunk 1]
```diff
@@ -64,6 +64,19 @@ FileComparison path_full_compare(char_u *const s1, char_u *const s2,
   expand_env(s1, exp1, MAXPATHL);
   bool id_ok_1 = os_fileid((char *)exp1, &file_id_1);
   bool id_ok_2 = os_fileid((char *)s2, &file_id_2);
+  void remove_spaces (char* restrict str_trimmed, const char* restrict str_untrimmed)
```
Comment (janlazo) [inline src/nvim/path.c]: When did C support nested functions?

**Comment by justinmk** [comment]
Marked "[WIP]", just rename to "[RFC]" when it's ready for review.

File: src/nvim/ex_cmds.c [hunk 2]
```diff
@@ -2201,7 +2202,12 @@ int do_ecmd(
             | ((flags & ECMD_FORCEIT) ? CCGD_FORCEIT : 0)
             | (eap == NULL ? 0 : CCGD_EXCMD))) {
     if (fnum == 0 && other_file && ffname != NULL)
-      (void)setaltfname(ffname, sfname, newlnum < 0 ? 0 : newlnum);
+      savingpath = (void)setaltfname(ffname, sfname, newlnum < 0 ? 0 : newlnum);
+      for (int i = 0; i < strlen(savingpath); i++) {
+        if (savingpath[i] == '/' && savingpath[i+1] == '/') {
+          savingpath[i] = savingpath[i+1] = '';
```
[the remark this answers is not in the document]
Comment (ZyX-I) [reply src/nvim/ex_cmds.c]: This line is outright incorrect, you can’t remove symbol by assigning `''` and even if you did this code removes two symbols. Also `strlen(savingpath)` must not be in cycle condition, you do such things by checking `savingpath[i]` for being zero. Assigning `void` to `char *` is never going to work too, I can see this forcing compiler to error out even without other errors (though I have not actually checked CI logs).

And as I said in original issue there still is at least one case when double slash *must* be preserved in what is a regular path (not handled by `*Cmd` commands). Also not familiar with code enough to say whether files like `http://…` will trigger this code, but they definitely must not. And it does not look like a correct location for the code for regular paths either, what is displayed in statusline has something to do with `buf->b_*fname` and this is not where they are defined.

**Comment by justinmk** [comment]
See #8862.

Thanks for the effort, but since this stalled let's forget it.

**Pull request closed without merging**